### DISCUSSION TITLE
How should I update IP entries within the ip-restriction?

### DISCUSSION THREAD

**Discussion by Chting** [Kong/kong#11303] [Help] [answered]
I use
curl -X POST http://kong:8001/plugins/ --data "name=ip-restriction"  --data "config.blacklist=x.x.x.x" 
Created a plugin，This is also effective，How should I use CURL to continue adding more IP to  blacklist ?
Also, how many IPs does this plugin support at most ?

kong version : 1.0.3

Thanks

**Comment by chronolaw** [comment]
Kong 1.0.3 is too old, I think you should upgrade to the latest 3.x version.
In Kong 3.x we should use `config.deny` or `config.allow`, which are array and can contain unlimited elements.

The doc is here. https://docs.konghq.com/hub/kong-inc/ip-restriction/

**Reply by Chting** [reply]
Yes, I know it's old, but it's troublesome, so I didn't upgrade.
I have read this document and it seems that there are no two questions I need here. Am I blind。

**Comment by chronolaw** [comment]
I did not read the code of Kong 1.x (it's too old), but I think that `config.blacklist` should work as `config.deny`.

You can add more ips by `-data "config.blacklist=x.x.x.x"`, give it a try.

**Reply by Chting** [reply]
Thank you for your reply。
The focus is on the front，I tried
curl -i -X PATCH --url http://kong:8001/plugins/0593265f-20bf-4da2-8ae5-196eafb9a6f5 --data "config.blacklist=XXXX"
It will change the value of the blacklist instead of continuing to add。

**Comment by chronolaw** [comment, marked as answer]
We can only change the ip list but not append it.

First, we should get the uuid of the plugin, like:
```
curl 127.1:8001/services/ngx-svc/plugins |jq
...
"id": "6a19f86e-9eaf-4c3a-b5f2-8a17e59b32ad",
...
```
Then we can use `PUT` or `PATCH` to modify the plugin:
```
curl -X PUT 127.1:8001/plugins/6a19f86e-9eaf-4c3a-b5f2-8a17e59b32ad \
  -d name=ip-restriction \
  -d config.deny=127.0.0.8 \
  -d config.deny=127.0.0.9 \
  |jq
```
It works with Kong 3.x, but I am not sure about Kong 1.x.

**Reply by Chting** [reply]
Okay, I understand. They should be universal, except for config.deny and config.blacklist
Thanks
